### ISSUE TITLE
Unable to change geographic location : UK example

### ISSUE THREAD

**Issue by meghanaponnuru123** [synthetichealth/synthea#640]
While I tried to run the example using country_code as UK instead of US and changed demographics , zipcodes and hospitals file I got the following error.

Running with options:
Population: 1
Seed: 0
Location: Shrewsbury, West Midlands
Min Age: 0
Max Age: 140
java.lang.NullPointerException
        at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
        at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
        at org.mitre.synthea.world.agents.Person.setProvider(Person.java:406)
        at org.mitre.synthea.world.agents.Person.setProvider(Person.java:411)
        at org.mitre.synthea.world.agents.Person.getProvider(Person.java:399)
        at org.mitre.synthea.world.agents.Person.encounterStart(Person.java:336)
        at org.mitre.synthea.modules.EncounterModule.process(EncounterModule.java:62)
        at org.mitre.synthea.engine.Generator.generatePerson(Generator.java:312)
        at org.mitre.synthea.engine.Generator.lambda$run$2(Generator.java:233)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
{alive=0, dead=0}

**Comment by jawalonoski** [comment]
Related to #629

**Comment by jawalonoski** [comment]
I've updated the Other Areas wiki.... but right now it won't work on the `master` branch. Checkout the `other_area_fixes` branch if you want to try it out. At least until I merge the fixes.

**Comment by meghanaponnuru123** [comment]
Sure, would check it. Thank you.

**Comment by AayushaY** [comment]
Hi Jawalonoski,

I tried out the other_area_fixes branch, unfortunately still doesn't work. Received the following error:

java.lang.RuntimeException: ERROR: Government Payer 'Medicare' does not exist.
        at org.mitre.synthea.world.agents.Payer.getGovernmentPayer(Payer.java:264)
        at org.mitre.synthea.modules.HealthInsuranceModule.determineInsurance(HealthInsuranceModule.java:80)
        at org.mitre.synthea.modules.HealthInsuranceModule.process(HealthInsuranceModule.java:54)
        at org.mitre.synthea.engine.Generator.generatePerson(Generator.java:342)
        at org.mitre.synthea.engine.Generator.lambda$run$2(Generator.java:259)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:835)

Thanks!

**Comment by jawalonoski** [comment]
@AayushaY can you confirm that you also made these changes the `other_area_fixes` branch: https://github.com/synthetichealth/synthea/wiki/Other-Areas#payer-data?

**Comment by AayushaY** [comment]
@jawalonoski 

Yes I did. Please find a screenshot of the payer info attached herewith. 

![Error](https://user-images.githubusercontent.com/41651705/73192096-31325e00-40ee-11ea-8b4e-a05cf245e34e.PNG)

Perhaps, this is worth bringing up - the config properties related to National Health Service seem to be unused by the code.

**Comment by AayushaY** [comment]
And yes, I made the necessary changes to demographics, zipcodes, timezones, hospitals, primary care facilities and country code as well. Please let me know if there's more changes I need to make to the code. 

Thank you!

**Comment by AayushaY** [comment]
@meghanaponnuru123 Can you confirm if the new fixed branch is working for you?

**Comment by jawalonoski** [comment]
@AayushaY Yes, they are used here:

https://github.com/synthetichealth/synthea/pull/643/files#diff-b10e8475dcc8721bc2592eb654b3ea2f

Are there somewhere else it should be used? (Other than the unit tests)

**Comment by jawalonoski** [comment]
I made temporary changes to the properties file in the branch, that work locally for me (though the unit tests break because they depend on the state of Massachusetts).

```
./run_synthea "West Midlands"
```

Eventually, I'll get around to fixing those unit test dependencies on Massachusetts...

**Comment by jawalonoski** [comment]
Looks like I forgot to handle urgent care facilities. More changes to come....

**Comment by AayushaY** [comment]
I was able to generate European patients. Thank you!

Quick question, is there a way to generate synthetic patients such that the patient's origin is from West Midlands, Europe but the provider is in the US. I tried changing the provider config properties to US hospitals for both default and primary care but I received the following error :

java.lang.NullPointerException
        at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHash
Map.java:1011)
        at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap
.java:1006)
        at org.mitre.synthea.world.agents.Person.setProvider(Person.java:559)
        at org.mitre.synthea.world.agents.Person.setProvider(Person.java:564)
        at org.mitre.synthea.world.agents.Person.getProvider(Person.java:552)
        at org.mitre.synthea.world.agents.Person.encounterStart(Person.java:474)
        at org.mitre.synthea.modules.EncounterModule.createEncounter(EncounterMod
ule.java:133)
        at org.mitre.synthea.modules.EncounterModule.process(EncounterModule.java
:64)
        at org.mitre.synthea.engine.Generator.generatePerson(Generator.java:343)
        at org.mitre.synthea.engine.Generator.lambda$run$2(Generator.java:259)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPool
Executor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoo
lExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:835)


Perhaps other changes in the code need to be made apart from just editing the config property. Please let me know what you think!

**Comment by jawalonoski** [comment]
When you say "the provider is in the US", do you mean in the export data the address shows up as "US" or it selected a provider actually in the United States?

**Comment by AayushaY** [comment]
I am trying to generate a FhirR4 object such that a patient from the UK comes to visit the US for a hospital visit. How would you reflect that in the code?

**Comment by AayushaY** [comment]
So the patient's address would be from the UK but the provider hospital should actually be in the United States. Does that make sense? :)

**Comment by jawalonoski** [comment]
Oh, I see. Synthea patients do not travel internationally, despite the fact they occasionally have passports. If you want to do that, we should open up another issue, and keep this one to domestic patients (for other areas).

**Comment by AayushaY** [comment]
Sounds good. I will open another issue. Thank you!

**Comment by scivm** [comment]
I am trying to test the uk branch

```
xxshambm@WL362836:~/git/synthea2$ git branch
  master
* other_area_fixes
```

I haven't changed anything but get a null pointer about half the time I execute the command 

`./run_synthea "West Midlands"`

```
Location: West Midlands
Min Age: 0
Max Age: 140
java.lang.NullPointerException
        at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
        at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
        at org.mitre.synthea.world.agents.Person.setProvider(Person.java:559)
        at org.mitre.synthea.world.agents.Person.setProvider(Person.java:564)
        at org.mitre.synthea.world.agents.Person.getProvider(Person.java:552)
        at org.mitre.synthea.world.agents.Person.encounterStart(Person.java:474)
        at org.mitre.synthea.modules.EncounterModule.createEncounter(EncounterModule.java:133)
        at org.mitre.synthea.modules.EncounterModule.process(EncounterModule.java:89)
        at org.mitre.synthea.engine.Generator.generatePerson(Generator.java:343)
        at org.mitre.synthea.engine.Generator.lambda$run$2(Generator.java:259)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
SLF4J: Class path contains multiple SLF4J bindings.
```

and when it works:

```
Min Age: 0
Max Age: 140
1 -- Osvaldo336 Russel238 (10 y/o M) Shrewsbury, West Midlands
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/xxshambm/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-log4j12/1.7.25/110cefe2df103412849d72ef7a67e4e91e4266b4/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/xxshambm/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-slf4j-impl/2.1/fe9d0925aeee68b743e9ea4b68ca9190a2a411a/log4j-slf4j-impl-2.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
{alive=1, dead=0}
```

**Comment by jawalonoski** [comment]
Yes, as I mentioned earlier in the discussion above, in that branch I forgot that we also need to update the urgent care facilities... I just did it.

The unit tests should fail (because they are hard-coded for the US) but you should be able to run the program without exceptions.

**Comment by scivm** [comment]
thanks.  I wonder if there is a way that people can contribute there country files that makes sense long term.  I'm working on Finland, Sweden and Norway.

**Comment by jawalonoski** [comment]
@scivm Yes! Please!!!

My ultimate goal is to support generating data for the entire world. I cannot do that without contributions on country files like the ones you are working on...

I'm working (slowly, in the background, in spare time) on modifying the unit tests to be location agnostic, but we're not quite there yet.

**Comment by scivm** [comment]
OK, sounds good.

One other file I modified was the names.yml.   I did it by updating english
section so far.   It likely should be separate files like the others since
my file is 300k with just Finnish first and last names.  Real names is
interesting because in Finland for example, many people have same last name
so when you build a patient interface you need to make sure you handle the
paging properly which you can test with a synthetic dataset.

For example:
M: [Juhani,Olavi,Antero,Tapani,Johannes,Tapio,Kalevi,Mikael
 family: [Korhonen,Virtanen,Mäkinen,Nieminen,Mäkelä,Hämäläinen

**Comment by scivm** [comment]
@jawalonoski Would synthea team consider adding a country code to the patient table?

**Comment by scivm** [comment]
Maybe there could be seperate git repositories that can be used to overlay the resources folder?  I think once this gets going and there are country contributions the repo size will grow out of control.

**Comment by jawalonoski** [comment]
> Would synthea team consider adding a country code to the patient table?

Yes.

> Maybe there could be seperate git repository for international work that can be used to overlay the resources folder with a static library of the synthea code? I think once this gets going and there are country contributions the java source code repo size will grow out of control.

Agreed. I will create another repository, called `synthea-international` or something like that, once I get this `other_area_fixes` branch passing the unit tests.

**Comment by jawalonoski** [comment]
@scivm Please see https://github.com/synthetichealth/synthea-international as a repository where we can accumulate international resource folders.

**Comment by scivm** [comment]
Ok.  I will send a pull request this weekend.

**Comment by andyt530** [comment]
Hi,

Just clones this branch and copied into place the stuff from https://github.com/synthetichealth/synthea-international

Like mentioned above it still has a problem with medicade

Min Age: 0
Max Age: 140
java.lang.RuntimeException: ERROR: Government Payer 'Medicare' does not exist.
        at org.mitre.synthea.world.agents.Payer.getGovernmentPayer(Payer.java:264)
        at org.mitre.synthea.modules.HealthInsuranceModule.determineInsurance(HealthInsuranceModule.java:80)
        at org.mitre.synthea.modules.HealthInsuranceModule.process(HealthInsuranceModule.java:54)
        at org.mitre.synthea.engine.Generator.generatePerson(Generator.java:342)
        at org.mitre.synthea.engine.Generator.lambda$run$2(Generator.java:259)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
SLF4J: Class path contains multiple SLF4J bindings.

The synthea.properties file has:

# Payers
generate.payers.insurance_companies.default_file = payers/insurance_companies_uk.csv
generate.payers.insurance_companies.medicare = National Health Service
generate.payers.insurance_companies.medicaid = National Health Service
generate.payers.insurance_companies.dual_eligible = National Health Service

---------------------------------------------------

 I had to change the insurance_companies_uk.csv to:

,id,name,address,city,state_headquartered,zip,phone,states_covered,services_covered,deductible,default_coinsurance,default_copay,monthly_premium,ownership
0,10001,National Health Service,Richmond House 79 Whitehall,Westminster,Greater London,SW1A 2NS,111,*,*,0,0.0,0,0,Government
1,10002,Medicare,Richmond House 79 Whitehall,Westminster,Greater London,SW1A 2NS,111,*,*,0,0.0,0,0,Government
2,10003,Medicaid,Richmond House 79 Whitehall,Westminster,Greater London,SW1A 2NS,111,*,*,0,0.0,0,0,Government
3,10004,Dual Eligible,Richmond House 79 Whitehall,Westminster,Greater London,SW1A 2NS,111,*,*,0,0.0,0,0,Government

as a workaround :-)

**Comment by jawalonoski** [comment]
@andyt530 Thank you for the verification, appreciated. I haven't finished making changes for `other_area_fixes` branch yet. More changes will be forthcoming.

**Comment by jawalonoski** [comment]
This should be fixed now with the merging of #643
